File: code/1-50/10.cpp
#include <bits/stdc++.h>

const int MAXN = 2E6;

std::vector<int> Prime;
std::bitset<MAXN> isnPri;

int main() {
	long long sum = 0;
	for (int i = 2; i < MAXN; ++i) {
		if (!isnPri[i]) {
			Prime.push_back(i);
			sum += i;
			// std::cout << i << "\n";
		}
		for (size_t j = 0; j < Prime.size(); ++j) {
			int cur = i * Prime[j];
			if (cur >= MAXN)
				break;
			isnPri[cur] = 1;
			if (i % Prime[j] == 0) break;
		}
	}
	std::cout << sum << "\n";
	return 0;
}

File: code/1-50/10.py
from sympy import primerange

def Euler(n):
	prime = []
	isnPri = [0] * n
	for i in range(2, n):
		if not isnPri[i]:
			prime.append(i)
		for j in prime:
			x = i * j
			if x >= n: break
			isnPri[x] = 1
			if i % j == 0: break
	return prime

if __name__ == '__main__':
	print(sum(Euler(2 * int(10 ** 6))))
	#print(sum(list(primerange(2, 2 * int(10 ** 6)))))